File: sum-linked-list/SumLinkedList.cs
using System;

public class SumLinkedList {
    public class Node {
        public Node next;
        public int value;
        public Node(int value) {
            this.value = value;
            this.next = null;
        }
    }

    public Node SumLinkedLists(Node linkedListOne, Node linkedListTwo) {
        Node linkedListHead = new Node(0);
        Node current = linkedListHead;
        int carry = 0;
        while(linkedListOne != null || linkedListTwo != null || carry != 0) {
            int one = (linkedListOne != null) ? linkedListOne.value : 0 ;
            int two = (linkedListTwo != null) ? linkedListTwo.value : 0 ;
            int sum = one + two + carry;

            Node newNode = new Node(sum % 10);
            current.next = newNode;
            current = newNode;

            carry = sum / 10;

            linkedListOne = (linkedListOne != null) ? linkedListOne.next : null;
            linkedListTwo = (linkedListTwo != null) ? linkedListTwo.next : null;
        }
        return linkedListHead.next;
    }
}

File: sum-linked-list/SumLinkedList.js
class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
      }
}
var sumLinkedList = (linkedListOne, linkedListTwo) => {
    let linkedListHead = new Node(0);
    let current = linkedListHead;
    let carryOver = 0;
    while(linkedListOne != null || linkedListTwo != null || carryOver != 0) {
        let one = (linkedListOne != null) ? linkedListOne.value : 0;
        let two = (linkedListTwo != null) ? linkedListTwo.value : 0;
        let sum = one + two + carryOver;

        let newNode = new Node(sum % 10);
        current.next = newNode;
        current = newNode;

        carryOver = Math.floor(sum / 10);

        linkedListOne = (linkedListOne != null) ? linkedListOne.next : null;
        linkedListTwo = (linkedListTwo != null) ? linkedListTwo.next : null;
    }
    return linkedListHead.next;
}

exports.Node = Node;
exports.sumLinkedList = sumLinkedList;